From 8d5f342844e28f6b62d9a6fc4afff49481deec44 Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Mon, 30 Nov 2020 21:33:06 +0200
Subject: [PATCH] [SW-20154] pp/preboot: make base addresses platform specific

CFG/SPI/FIT/SRAM base addresses can vary between
different platforms. Hence move them to individual
platform specific folders.

Change-Id: I04f36624a0a82ca70f3859a103152fea7003a5ca
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/gaudi/common.h b/gaudi/common.h
index 127afec..6ae8d79 100644
--- a/gaudi/common.h
+++ b/gaudi/common.h
@@ -7,8 +7,6 @@
 #ifndef COMMON_GAUDI_H_
 #define COMMON_GAUDI_H_
 
-#include <preboot.h>
-
 #define PPBOOT_SIZE			0x4000		/* 16KB */
 #define PREBOOT_SIZE			0xC000		/* 48KB */
 #define SRAM_SIZE			0x1400000	/* 20MB */
@@ -24,6 +22,11 @@
 #define PPBOOT_TEXT_BASE		0x7FF0000000
 #define PREBOOT_TEXT_BASE		(PPBOOT_TEXT_BASE + PPBOOT_SIZE)
 
+#define HL_SPI_BASE			0x7FF8000000ull
+#define HL_CFG_BASE			0x7FFC000000ull
+#define SDRAM_BASE			0x7FF0000000ull
+#define FIT_IMG_BASE			0x7FF0200000ull
+
 #define FUSE_WORDS_PER_BANK		64
 #define I2C_SLAVE_ADDRESS		0x4f
 #define I2C_SLAVE_ADDRESS_DBG		0x47
diff --git a/goya/common.h b/goya/common.h
index 67e2c20..b6cb1ad 100644
--- a/goya/common.h
+++ b/goya/common.h
@@ -7,8 +7,6 @@
 #ifndef COMMON_GOYA_H_
 #define COMMON_GOYA_H_
 
-#include <preboot.h>
-
 #define SRAM_SIZE			0x32A0000	/* 50.625MB */
 
 #define PPBOOT_SIZE			0x4000		/* 16KB */
@@ -26,6 +24,11 @@
 
 #define BOOT_INFO_BASE		HL_SP_SRAM_BASE
 
+#define HL_SPI_BASE			0x7FF8000000ull
+#define HL_CFG_BASE			0x7FFC000000ull
+#define SDRAM_BASE			0x7FF0000000ull
+#define FIT_IMG_BASE			0x7FF0200000ull
+
 #define FUSE_WORDS_PER_BANK	0x10
 #define I2C_SLAVE_ADDRESS	0x4f
 #define I2C_SLAVE_ADDRESS_DBG	0x47
diff --git a/goya2/common.h b/goya2/common.h
index b79b86f..f9ed2ae 100644
--- a/goya2/common.h
+++ b/goya2/common.h
@@ -6,7 +6,6 @@
 #ifndef COMMON_GOYA2_H_
 #define COMMON_GOYA2_H_
 
-#include <preboot.h>
 #include <goya2_blocks_linux_driver.h>
 #include <goya2/goya2.h>
 
@@ -29,6 +28,11 @@
  */
 #define BOOT_INFO_BASE			HL_SP_SRAM_BASE
 
+#define HL_SPI_BASE			0x7FF8000000ull
+#define HL_CFG_BASE			0x7FFC000000ull
+#define SDRAM_BASE			0x7FF0000000ull
+#define FIT_IMG_BASE			0x7FF0200000ull
+
 #define FUSE_WORDS_PER_BANK		0x40
 #define I2C_SLAVE_ADDRESS		0x6A
 #define I2C_SLAVE_ADDRESS_DBG		0x47
diff --git a/goya2_arc/common.h b/goya2_arc/common.h
index 58a5ffd..64426a1 100644
--- a/goya2_arc/common.h
+++ b/goya2_arc/common.h
@@ -7,7 +7,6 @@
 #ifndef COMMON_GOYA2_ARC_H_
 #define COMMON_GOYA2_ARC_H_
 
-#include <preboot.h>
 #include <goya2_blocks_linux_driver.h>
 
 #define PPBOOT_SIZE		0x4000		/* 16KB */
@@ -29,6 +28,11 @@
  */
 #define PPBOOT_DCCM_BASE	0x80002000
 
+#define SDRAM_BASE		0x10000000
+#define FIT_IMG_BASE		0x10200000
+#define HL_SPI_BASE		0x1A000000
+#define HL_CFG_BASE		0xFC000000
+
 #define FUSE_WORDS_PER_BANK	0x40
 #define I2C_SLAVE_ADDRESS	0x6A
 #define I2C_SLAVE_ADDRESS_DBG	I2C_SLAVE_ADDRESS
diff --git a/goya2_arc_gnu/common.h b/goya2_arc_gnu/common.h
index cd310f2..336003d 100644
--- a/goya2_arc_gnu/common.h
+++ b/goya2_arc_gnu/common.h
@@ -6,7 +6,6 @@
 #ifndef COMMON_GOYA2_ARC_GNU_H_
 #define COMMON_GOYA2_ARC_GNU_H_
 
-#include <preboot.h>
 #include <goya2_blocks_linux_driver.h>
 
 #define SRAM_SIZE			0x8000000	/* 128MB */
@@ -38,6 +37,11 @@
 #define RECOVERY_TEXT_BASE	(0x0 + 0x4)
 #define RECOVERY_SIZE		0x4000
 
+#define SDRAM_BASE		0x10000000
+#define FIT_IMG_BASE		0x10200000
+#define HL_SPI_BASE		0x1A000000
+#define HL_CFG_BASE		0xFC000000
+
 /*
  * extracts "num" number of bits from position "pos" in the
  * given number "value"
diff --git a/include/preboot.h b/include/preboot.h
index 3c043b6..035e92f 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -8,24 +8,16 @@
 #ifndef COMMON_PREBOOT_H_
 #define COMMON_PREBOOT_H_
 
+#include <common.h>
+
 #define BIT(nr)					(1 << (nr))
 #define ARRAY_SIZE(x)				(sizeof(x) / sizeof((x)[0]))
 
 #ifdef __arc__
-#define HL_SPI_BASE				0x1A000000
-#define HL_CFG_BASE				0xFC000000
-#define SDRAM_BASE				0x10000000
-#define FIT_IMG_BASE				0x10200000
-
 #define wfe()	__asm__ __volatile__("sleep 0x11\n\t" : : )
 #endif
 
 #ifdef __aarch64__
-#define HL_SPI_BASE				0x7FF8000000ull
-#define HL_CFG_BASE				0x7FFC000000ull
-#define SDRAM_BASE				0x7FF0000000ull
-#define FIT_IMG_BASE				0x7FF0200000ull
-
 #define wfe()	__asm__ __volatile__ ("wfe\n\t" : : )
 #endif
 
diff --git a/include/sha256.h b/include/sha256.h
index 539fdde..fe588cc 100644
--- a/include/sha256.h
+++ b/include/sha256.h
@@ -5,6 +5,8 @@
 #ifndef _SHA256_H
 #define _SHA256_H
 
+#include <preboot.h>
+
 #define SHA256_SUM_LEN	32
 #define SHA256_DER_LEN	19
 
diff --git a/preboot/fit_rsa.h b/preboot/fit_rsa.h
index 6adeeac..be94aca 100644
--- a/preboot/fit_rsa.h
+++ b/preboot/fit_rsa.h
@@ -6,6 +6,8 @@
 #ifndef COMMON_FIT_RSA_H_
 #define COMMON_FIT_RSA_H_
 
+#include <preboot.h>
+
 struct rsa_key_prop {
 	const void *rr;			/* R^2 can be treated as byte array */
 	const void *modulus;		/* modulus as byte array */
